lib/Auth/Process/COmanageDbClient.php

```diff
@@ -323,8 +324,18 @@ public function process(&$state)
                   $callback = SimpleSAML_Module::getModuleURL('attrauthcomanage/idp_callback.php', array('stateId' => $id));
                   SimpleSAML_Logger::debug("[attrauthcomanage] process: callback url => " . $callback);
                   $params = array("targetnew" => $callback);
+                  // Check if community signup is required
                   if (!empty($state['saml:AuthenticatingAuthority']) && in_array(end($state['saml:AuthenticatingAuthority']), $this->communityIdps, true)) {
-                    \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
+                      // Redirect to community signup flow with all
+                      // attributes available including affiliation
+                      if (!empty($state['Attributes']['voPersonExternalAffiliation'])
+                          && !empty($state['Attributes']['mail'])
+                          && !empty($state['Attributes']['givenName'])
+                          && !empty($state['Attributes']['sn'])) {
+                          \SimpleSAML\Utils\HTTP::redirectTrustedURL($this->registryUrls['community_sign_up'], $params);
+                      } elseif (!empty($this->registryUrls['community_sign_up_no_aff'])) {
```

Comment:
if the `community_sign_up_no_aff` is not defined it should still redirect to the `community_sign_up` which is mandatory